### PR TITLE
Improve the comments in the header-only template

### PR DESCRIPTION
**Pull request by RubenRBS**
Found some things in the header-only template that could use a rewording to better understand their message when I was playing around with this, so I changed some of those comments.

This also removes one important comment as a result of #14845

### CODE REVIEW

File: docs/package_templates/header_only/all/conanfile.py [hunk 1]
```diff
@@ -14,21 +14,23 @@ class PackageConan(ConanFile):
...
     url = "https://github.com/conan-io/conan-center-index"
     homepage = "https://github.com/project/package"
-    # no "conan" and project name in topics. Use topics from the upstream listed on GH
-    # Keep 'hearder-only' as topic
+    # Do not put "conan" nor the project name in topics. Use topics from the upstream listed on GH
+    # Keep 'header-only' as topic
     topics = ("topic1", "topic2", "topic3", "header-only")
-    settings = "os", "arch", "compiler", "build_type"  # even for header only
-    no_copy_source = True  # do not copy sources to build folder for header only projects, unless, need to apply patches
+    # Keep these unless you have a good reason
```
[the remark this answers is not in the document]
Comment (prince-chrismc) [reply docs/package_templates/header_only/all/conanfile.py:24]: This is solid IMO 👍 

```suggestion
    # Keep these unless you can explain why it's not required for this particular case
```

**Comment by prince-chrismc** [comment]
You'll need to retrigger CI, close the pr wait 10s and then re-open it 🔁

File: docs/package_templates/header_only/all/conanfile.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (RubenRBS) [reply docs/package_templates/header_only/all/conanfile.py:24]: Very good idea, I'll try to apply the same reasoning for all my changes in the future!

**Review by uilianries — APPROVED** [review]
LGTM

**Review by prince-chrismc — APPROVED** [review]

**Pull request merged**